### PR TITLE
Use internal IP whilst in GCE

### PR DESCRIPTION
**Pull request by ravwojdyla**
Fixes: #229

FYI @eric-czech

### CODE REVIEW

**Comment by eric-czech** [comment]
Makes sense to me, thanks @ravwojdyla!

File: dask_cloudprovider/gcp/utils.py [hunk 1]
```diff
@@ -12,3 +12,22 @@ def inner(http, *args, **kwargs):
         return googleapiclient.http.HttpRequest(new_http, *args, **kwargs)
 
     return inner
+
+
+def is_gce_env() -> bool:
```
Comment (jacobtomlinson) [inline dask_cloudprovider/gcp/utils.py:17]: I'm not sure about this function name. I don't think it's immediately obvious what it does.

What about `is_inside_gce` instead?

File: dask_cloudprovider/gcp/instances.py [hunk 2]
```diff
@@ -293,8 +292,16 @@ async def start_scheduler(self):
         self.internal_ip, self.external_ip = await self.create_vm()
 
         if self.config.get("public_ingress", True):
-            # scheduler is publicly available
-            self.address = f"tcp://{self.external_ip}:8786"
+            if is_gce_env():
+                # if the client is running inside GCE environment
+                # it's better to use internal IP, which doesn't
+                # require firewall setup
+                self.address = f"tcp://{self.internal_ip}:8786"
```
Comment (jacobtomlinson) [inline dask_cloudprovider/gcp/instances.py:299]: We have some duplication here. Can we combine this with the outer if statement.

**Review by jacobtomlinson — COMMENTED** [review]
Many thanks for this. I agree this change would be very valuable.

Just a couple of small comments.

File: dask_cloudprovider/gcp/utils.py (hunk 1 above)
Comment (ravwojdyla) [reply dask_cloudprovider/gcp/utils.py]: Sure - renamed.

File: dask_cloudprovider/gcp/instances.py (hunk 2 above)
Comment (ravwojdyla) [reply dask_cloudprovider/gcp/instances.py]: Good point, fixed.

File: dask_cloudprovider/gcp/utils.py (hunk 1 above)
Comment (jacobtomlinson) [reply dask_cloudprovider/gcp/utils.py]: Thanks!

**Comment by jacobtomlinson** [comment]
Thanks for the fast review feedback turnaround!

Awesome work!

**Pull request merged**